libnavui-androidauto/src/main/java/com/mapbox/androidauto/car/preview/CarRoutePreviewScreen.kt

```diff
@@ -181,10 +182,12 @@ class CarRoutePreviewScreen(
                 Action.Builder()
                     .setTitle(carContext.getString(R.string.car_action_preview_navigate_button))
                     .setOnClickListener {
-                        routePreviewCarContext.mapboxNavigation.setNavigationRoutes(
-                            carRoutesProvider.navigationRoutes.value
-                        )
-                        MapboxCarApp.updateCarAppState(ActiveGuidanceState)
+                        MapboxNavigationApp.current()?.let { mapboxNavigation ->
```

Comment:
Looks like it is important to have `mapboxNavigation` here, should we use `!!` instead? Or at least log a warning.

Reply:
yeah.. thinking about this a bit and going with `!!`. Going to list a couple examples where a developer could create the crash. In particular, this crash could only happen because of `3`. It is a small window of a crash, but we could potentially cure the `3` case with a `reset` function 🤔.

1. If you `current()!!` before the `setup`
``` kotlin
MapboxNavigationApp.current()!!.setNavigationRoutes(emptyList())
MapboxNavigationApp.setup(..)
```

2. If you `current()!!` before the `attach` lifecycle `onCreate` is called
``` kotlin
MapboxNavigationApp.attach(..)
MapboxNavigationApp.current()!!.setNavigationRoutes(emptyList())
```

3. When you're in the middle of resetting options
``` kotlin
val navigationOption = MapboxNavigationApp.current()!!.navigationOptions
MapboxNavigationApp.disable()
MapboxNavigationApp.current()!!.setNavigationRoutes(..)
MapboxNavigationApp.setup(navigationOption)
```

Reply:
Will merge here, but I think we should consider adding a ReadOnlyDelegate that makes it so `current()!!` is not a common practice

Reply:
Here's the `ReadOnlyDelegate` https://github.com/mapbox/mapbox-navigation-android/pull/6233 😄